src/cell_group.hpp

```diff
@@ -231,7 +241,7 @@ class cell_group {
         auto handle = get_probe_handle(probe_id);
 
         auto sampler_index = uint32_t(samplers_.size());
-        samplers_.push_back({handle, s});
+        samplers_.push_back({handle, probe_id.gid, s});
```

Comment:
The interface of this constructor would be nicer if only the probe_id would be given (instead of the gid).